src/smartactions/SchedulingBase.sol

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import "modulekit/core/sessionKey/ISessionValidationModule.sol";
+import { IERC20 } from "forge-std/interfaces/IERC20.sol";
+import { IERC7579Execution } from "modulekit/Accounts.sol";
+import { ERC7579ExecutorBase } from "modulekit/Modules.sol";
+import "modulekit/core/sessionKey/ISessionValidationModule.sol";
+
+abstract contract SchedulingBase is ERC7579ExecutorBase, ISessionValidationModule {
+    error InvalidExecution();
+    error InvalidMethod(bytes4);
+    error InvalidValue();
+    error InvalidAmount();
+    error InvalidTarget();
+    error InvalidRecipient();
+
+    error InvalidInstall();
+
+    error InvalidJob();
+
+    event ExecutionAdded(address indexed smartAccount, uint256 indexed jobId);
+
+    event ExecutionTriggered(address indexed smartAccount, uint256 indexed jobId);
+
+    event ExecutionStatusUpdated(address indexed smartAccount, uint256 indexed jobId);
+
+    mapping(address smartAccount => mapping(uint256 jobId => ExecutionConfig)) internal
```

Comment:
mapping is internal. as mentioned in the previous call.  we need a view function for this to query the contract on `ExecutionStatusUpdated`